src/smc-hub/hub.coffee

```diff
@@ -80,6 +80,15 @@ MetricsRecorder = require('./metrics-recorder')
 # express http server -- serves some static/dynamic endpoints
 hub_http_server = require('./hub_http_server')
 
+if fs.existsSync(path_module.join(SMC_ROOT, 'smc-hub', 'lti', 'lti.ts'))
```

Comment:
This seems weird to me.  How is this going to work?  Why not just use try/catch?   Can you put some comments in the code.    Also, what is the intention longterm?  I.e., is it that we copy an lti.ts file in at runtime?

Reply:
true, I changed that to try/catch.

... and basically yes, I treat the whole `smc-hub/lti` subdirectory as being optional and it's only the main hub file that knows anything about it. Deployment could e.g. mean to copy or git-clone this LTI module somewhere and then set a symlink.